src/radar/radar/quantum.py

```diff
@@ -1,58 +1,166 @@
+        
+        # loop until Raymarine Quantum is found
+        while True:
+            self.get_logger().info(f'Locating radar...')
+            data, senderaddr = locator_socket.recvfrom(1024)
+            self.get_logger().debug(f'received ({len(data)} bytes) from {senderaddr}')
+            if len(data) != 36: continue # ignore any packets not 36 bytes
             
-            # Tell the operating system to add the socket to the multicast group
-            # on all interfaces.
-            mreq = struct.pack('4sL', socket.inet_aton(li.data_ip), socket.INADDR_ANY)
-            report_socket.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, mreq)
-            logger.info(f'Initialized report_socket')
+            quantum_location = LocationInfo(*LocationInfo.parse(data))
+            self.get_logger().debug(f'{quantum_location=}')
+            if quantum_location.model_id != quantum_model_id: continue
             
-            logger.info(f'Run radar receive thread')
-            receive_thread = Thread(target = radar.receive.listen_radar, 
-                                    args=(report_socket,))
-            receive_thread.start()
-            receive_thread.join(SCAN_DURATION_SECONDS)
-            logger.info(f'Ending receive thread')
-        
-        radar.control.radar_tx_off(command_socket=command_socket,
-                                   command_ip=li.radar_ip,
-                                   command_port=li.radar_port)
-        keep_alive_thread.join(1)
+            self.get_logger().info(f'Found radar at {quantum_location.radar_ip}')
+            break
+        
+        self.get_logger().info(f'Closing locator socket')
+        locator_socket.close()
+        return quantum_location
+
+
+    def create_command_socket(self) -> socket:
+        command_socket = socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP)
+        command_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        command_socket.setblocking(0) # non-blocking socket
+        self.get_logger().info(f'Initialized command socket')
+        
+        return command_socket
+
+
+    def create_report_socket(self) -> socket:
+        report_socket = socket.socket(family=socket.AF_INET, type=socket.SOCK_DGRAM, proto=socket.IPPROTO_UDP)
+        report_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        # Bind to the server address
+        # on this port, receives ALL multicast groups
+        report_socket.bind(('', self.quantum_location.data_port))
+        # Tell the operating system to add the socket to the multicast group
+        # on all interfaces.
+        mreq = struct.pack('4sL', socket.inet_aton(self.quantum_location.data_ip), socket.INADDR_ANY)
+        report_socket.setsockopt(socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, mreq)
+        self.get_logger().info(f'Initialized report socket')
+        
+        return report_socket
```

Comment:
@Ryanf55
Running openCPN also creates the same socket listening on 0.0.0.0:2574 according to the `ss` command. BTW small correction: 2574 is the radar data receive multicast and 2575 is the radar command unicast (I had the port numbers backwards during the meeting).

Current hypothesis is that that one of the lines above initializing the socket is silently failing.